botorch/optim/optimize_mixed.py

```diff
@@ -676,22 +773,29 @@ def optimize_acqf_mixed_alternating(
     tkwargs: dict[str, Any] = {"device": bounds.device, "dtype": bounds.dtype}
     # Remove fixed features from dims, so they don't get optimized.
     discrete_dims = [dim for dim in discrete_dims if dim not in fixed_features]
-    if len(discrete_dims) == 0:
+    cat_dims = cat_dims or []
+    cat_dims = [dim for dim in cat_dims if dim not in fixed_features]
+    non_cont_dims = [*discrete_dims, *cat_dims]
```

Comment:
we should probably validate that `cat_dims` and `discrete_dims` are disjoint here

Reply:
nvm I guess we're doing that below